rpc/cloudagent/network/publicipaddress/moc_cloudagent_publicipaddress.proto

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+syntax = "proto3";
+option go_package = "github.com/microsoft/moc/rpc/cloudagent/network";
+package moc.cloudagent.network;
+
+import "google/protobuf/wrappers.proto";
+import "moc_common_common.proto";
+import "moc_common_networkcommon.proto";
+
+// ---------------------------------------------------------------------------------------------------
+// Azure public IP address overview
+// https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/virtual-network-public-ip-address
+// Azure CLI:
+// https://learn.microsoft.com/en-us/cli/azure/network/public-ip?view=azure-cli-latest
+// ---------------------------------------------------------------------------------------------------
+
+message PublicIPAddressRequest {
+	repeated PublicIPAddress PublicIPAddresses = 1;
+	Operation OperationType = 2;
+}
+
+message PublicIPAddressResponse {
+	repeated PublicIPAddress PublicIPAddresses = 1;
+	google.protobuf.BoolValue Result = 2;
+	string Error = 3;
+}
+
+message PublicIPAddressPrecheckRequest {
+	repeated PublicIPAddress PublicIPAddresses = 1;
+}
+
+message PublicIPAddressPrecheckResponse {
+	// The precheck result: true if the precheck criteria is passed; otherwise, false
+	google.protobuf.BoolValue Result = 1;
+
+	// The error message if the precheck is not passed; otherwise, empty string
+	string Error = 2;
+}
+
+message PublicIPAddress {
```

Comment:
@kybeak Please can we follow what Azure-SDK is following . Lets use the same struct. We have been doung the same for all the entities https://[raw.githubusercontent.com/Azure/azure-sdk-for-go/refs/tags/v68.0.0/services/network/mgmt/2022-07-01/network/models.go](https://raw.githubusercontent.com/Azure/azure-sdk-for-go/refs/tags/v68.0.0/services/network/mgmt/2022-07-01/network/models.go)

Reply:
@nijosmsft Are we good this ?

Reply:
@raghavendra-nataraj Kyung told me that you suggested to add *Properties. All parent resources in moc/azure sdk have a *Properties field and I haven't seen that pattern in protobuf as it is our internal implementation. Kyung was following the existing pattern in other moc entities. This is bit of churn to change it. But please let us know if this is a new requirement we are enforcing now.